packages/core/admin/admin/src/content-manager/hooks/useDragAndDrop.js

```diff
@@ -42,49 +43,102 @@ export const useDragAndDrop = (
+    canDropHandler,
   }
 ) => {
   const objectRef = useRef(null);
 
-  const [{ handlerId }, dropRef] = useDrop({
+  const [{ handlerId, canDrop }, dropRef] = useDrop({
     accept: type,
+    canDrop: canDropHandler, // to skip some items from droppable target
     collect(monitor) {
       return {
         handlerId: monitor.getHandlerId(),
+        canDrop: monitor.canDrop(),
       };
     },
     hover(item, monitor) {
       if (!objectRef.current) {
         return;
       }
-      const dragIndex = item.index;
-      const newInd = index;
 
-      // Don't replace items with themselves
-      if (dragIndex === newInd) {
+      // Don't move item if not allowed to drop at new index
+      if (!canDrop) {
         return;
       }
+      const dragIndex = item.index;
+      const newIndex = index;
 
-      if (dropSensitivity === 'regular') {
-        const hoverBoundingRect = objectRef.current.getBoundingClientRect();
-        const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
-        const clientOffset = monitor.getClientOffset();
-        const hoverClientY = clientOffset.y - hoverBoundingRect.top;
-
-        // Dragging downwards
-        if (dragIndex < newInd && hoverClientY < hoverMiddleY) {
+      if (typeof index === 'number') {
+        if (dragIndex === newIndex) {
+          // Don't replace items with themselves
           return;
         }
 
-        // Dragging upwards
-        if (dragIndex > newInd && hoverClientY > hoverMiddleY) {
-          return;
+        if (dropSensitivity === 'regular') {
+          const hoverBoundingRect = objectRef.current.getBoundingClientRect();
+          const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+          const clientOffset = monitor.getClientOffset();
+          const hoverClientY = clientOffset.y - hoverBoundingRect.top;
+
+          // Dragging downwards
+          if (dragIndex < newIndex && hoverClientY < hoverMiddleY) {
+            return;
+          }
+
+          // Dragging upwards
+          if (dragIndex > newIndex && hoverClientY > hoverMiddleY) {
+            return;
+          }
         }
-      }
 
-      // Time to actually perform the action
-      onMoveItem(newInd, dragIndex);
-      item.index = newInd;
+        // Time to actually perform the action
+        onMoveItem(newIndex, dragIndex);
+        item.index = newIndex;
+      } else {
+        // Using numbers as indices don't work for heirarchy of nodes
+        // For moving blocks, index would be a path as Array<number>
+        if (dropSensitivity === 'regular') {
+          const hoverBoundingRect = objectRef.current.getBoundingClientRect();
+          const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+          const clientOffset = monitor.getClientOffset();
+          const hoverClientY = clientOffset.y - hoverBoundingRect.top;
+
+          // Indices comparison to find elments position in a heirarchy of nodes
```

Comment:
Everyone not working on blocks/slate will not understand this (incl. me) considering nodes is typically associated to the DOM